### ISSUE TITLE
[Feature Request] Support vim-lsp

### ISSUE THREAD

**Issue by watcol** [joshdick/onedark.vim#251]
[vim-lsp](https://github.com/prabirshrestha/vim-lsp), a LSP client for vim, has these hightlights for displaying diagnostics.

Signs on the left of line:
- `LspError`
- `LspWarning`
- `LspInformation`
- `LspHint`

Virtual texts on the right of line:
- `LspErrorVirtualText`
- `LspWarningVirtualText`
- `LspInformationVirtualText`
-  `LspHintVirtualText`

Text highlight:
- `LspErrorHighlight`
- `LspWarningHighlight`
- `LspInformationHighlight`
-  `LspHintHighlight`

May I ask you to support them?

For more resources, see [vim-lsp.txt](https://github.com/prabirshrestha/vim-lsp/blob/master/doc/vim-lsp.txt) or ask me on this chat.

**Comment by dagadbm** [comment]
yes pls :D

**Comment by joshdick** [comment]
I just merged #270 to add the basic highlight groups for signs. The virtual text groups should inherit from these, so I don't believe they're necessary.

I don't (yet!) use this plugin, so I'm not sure whether the text highlight groups are strictly necessary either?

**Comment by watcol** [comment]
First, I'm sorry that I've misunderstood that signs are highlighted with `LspError`, `LspWarning`, etc. Actually, they are highlighted with `LspErrorText`, and `LspError` makes no effect. Then the virtual text highlights are automatically linked to `LspErrorText`, I don't think they are necessary, either. But I expect `LspErrorHighlights` to underlined, like `ALEError`, I think they're necessary.

So I want you to support `LspErrorText`, `LspWarningText`, ... and `LspErrorHighlight`, `LspWarningHighlight`, ...

Thank you for your response.

**Comment by joshdick** [comment]
I know it's over a year later (!), but to follow up on this issue:

onedark.vim currently supports vim-lsp's "signs" highlight groups for `vim-lsp`: `LspError`, `LspWarning`...

vim-lsp's [documentation mentions](https://github.com/prabirshrestha/vim-lsp/blob/68c018e/doc/vim-lsp.txt#L646):

> Virtual text uses the same highlight groups used for signs (eg `LspErrorText`),

...but this mention of `LspErrorText` seems like a typo/holdover from a previous version of the plugin since signs are documented as using the groups that are currently in onedark.vim, and the rest of vim-lsp's documention doesn't mention any other `Lsp*Text` highlight groups.

So in summary, it looks like only the `Lsp*Highlight` groups need to be added to `onedark.vim`? Because I don't use this plugin, I can't tell if setting these to the same values as their existing sign counterparts is desirable/correct?

**Comment by watcol** [comment]
I answer your question in the first: the `Lsp*Highlight` has very similar behavior as `ALEError`, `ALEWarning`, ..., so I think setting the same values is desirable.

Then, I'm sorry if my former comments confused you, but `LspError`, `LspWarning`, ... are actually the name of **sign** groups (not highlight groups), referring to the documentation:
> Four groups of signs are defined and used:
> `LspError`, `LspWarning`, `LspInformation`, `LspHint`.

And I confirmed that `Lsp*Text` highlights signs in the latest version of vim-lsp (though it's not mentioned in the docs) by running `:sign list`:
```
sign LspError text=E> linehl=LspErrorLine texthl=LspErrorText
sign LspWarning text=W> linehl=LspWarningLine texthl=LspWarningText
...
```

So I think `Lsp*Text` should be supported, and `LspError`, `LspWarning`, ... could be removed, as I said before.

**Comment by joshdick** [comment]
OK, that makes sense after your latest explanation.

I just opened https://github.com/joshdick/onedark.vim/pull/308 (branch name: `fix-vim-lsp`) that I believe captures everything described in this issue.

@watcol Can you try it out and let me know if it looks/behaves as you expect?

Thanks!

**Comment by watcol** [comment]
It seems very well. Thank you! :smile:

**Comment by joshdick** [comment]
@watcol Merged, glad you like it, and sorry it took me so long!
